spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraTableScanRDD.scala

```diff
@@ -77,6 +78,21 @@ class CassandraTableScanRDD[R] private[connector](
 
   override type Self = CassandraTableScanRDD[R]
 
+  private def copy(numPartitions: Option[Int]): Self = {
+    new CassandraTableScanRDD[R](
+      sc = sc,
+      connector = connector,
+      keyspaceName = keyspaceName,
+      tableName = tableName,
+      columnNames = columnNames,
+      where = where,
+      limit = limit,
+      numPartitions = numPartitions,
```

Comment:
Could we just manipulate the read conf's splitCount rather than introducing a new variable here?